core/src/main/scala/org/apache/spark/scheduler/OutputCommitCoordinator.scala

```diff
@@ -48,25 +48,29 @@ private[spark] class OutputCommitCoordinator(conf: SparkConf, isDriver: Boolean)
   private type StageId = Int
   private type PartitionId = Int
   private type TaskAttemptNumber = Int
+  private case class StageState(
+      authorizedCommitters: Array[TaskAttemptNumber],
+      failures: mutable.Map[PartitionId, mutable.Set[TaskAttemptNumber]])
```

Comment:
Why not define failures as an member variable (and initialize it there with an empty map), rather than forcing the caller to pass in an empty map?